tensorrt_llm/_torch/pyexecutor/py_executor.py

```diff
@@ -923,37 +931,36 @@ def _executor_loop_overlap(self):
                         if r.get_context_remaining_length() == 0
                     ]
 
-                    self.previous_batch = (scheduled_batch, new_tensors_device,
-                                           new_tensors_host, decoder_event,
-                                           iter_start_time, iter_stats,
-                                           ctx_transmission_reqs)
-                    self._gather_dp_requests_num()
+                self.previous_batch = BatchState(
```

Comment:
This line is moved from the scope of `if scheduled_batch.batch_size > 0:`. Is it expected?

Reply:
Nice catch! Fixed.